x/clp/handler.go

```diff
@@ -0,0 +1,516 @@
+package clp
+
+import (
+	"fmt"
+	"github.com/Sifchain/sifnode/x/clp/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+// NewHandler creates an sdk.Handler for all the clp type messages
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+		switch msg := msg.(type) {
+		case MsgCreatePool:
+			return handleMsgCreatePool(ctx, k, msg)
+		case MsgDecommissionPool:
+			return handleMsgDecommissionPool(ctx, k, msg)
+		case MsgAddLiquidity:
+			return handleMsgAddLiquidity(ctx, k, msg)
+		case MsgRemoveLiquidity:
+			return handleMsgRemoveLiquidity(ctx, k, msg)
+		case MsgSwap:
+			return handleMsgSwap(ctx, k, msg)
+		default:
+			errMsg := fmt.Sprintf("unrecognized %s message type: %T", ModuleName, msg)
+			return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg)
+		}
+	}
+}
+
+func handleMsgDecommissionPool(ctx sdk.Context, keeper Keeper, msg MsgDecommissionPool) (*sdk.Result, error) {
+	// Verify pool
+	pool, err := keeper.GetPool(ctx, msg.Ticker)
+	if err != nil {
+		return nil, types.ErrPoolDoesNotExist
+	}
+	if pool.ExternalAssetBalance+pool.NativeAssetBalance > keeper.GetParams(ctx).MinCreatePoolThreshold {
```

Comment:
The min threshold is just a placeholder for now ,so that we have the plumbing to define it . 
Yes your point makes complete sense ,we do need to discuss this more

Reply:
@AustinoBombino , Can we also add this to the list of topics to discuss and document ?

Reply:
I think we should just have a minimum threshold of `pool.NativeAssetBalance` cc @Jedi2002 @AustinoBombino @musnit

Reply:
Asana task created to resolve this: https://app.asana.com/0/1196711589172801/1198992855386367/f

Reply:
@jkbrooks yep i think making it based purely on NativeAssetBalance works fine.